crates/storage/codegen/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+use syn::Data;
+
+/// Provides common methods for `DeriveInput`.
+///
+/// **Note:** This is only for internal usage in the `codegen` module.
+pub trait DeriveUtils {
+    /// Finds the salt of the structure, enum or union.
+    /// The salt is any generic that has bound `StorageKey`.
+    /// In most cases it is parent storage key or auto-generated storage key.
+    fn find_salt(&self) -> Option<syn::TypeParam>;
```

Comment:
After renaming in the trait, I added the comment here that it is "parent storage key or auto-generated storage key". Because in most cases, it is used as salt to generate the final storage key=)